src/main/java/com/conveyal/datatools/manager/models/OtpServer.java

```diff
@@ -91,4 +125,313 @@ public Project parentProject() {
     public void delete () {
         Persistence.servers.removeById(this.id);
     }
+
+    @JsonIgnore
+    @BsonIgnore
+    public AmazonEC2 getEC2Client() throws CheckedAWSException {
+        return EC2Utils.getEC2Client(role, getRegion());
+    }
+
+    @JsonIgnore
+    @BsonIgnore
+    private String getRegion() {
+        return ec2Info != null && ec2Info.region != null
+            ? ec2Info.region
+            : null;
+    }
+
+    /**
+     * Asynchronously validates all ec2 config of a the OtpServer instance.
+     */
+    @JsonIgnore
+    @BsonIgnore
+    public EC2ValidationResult validateEC2Config()
```

Comment:
Can we move the EC2 config validation somewhere else? By moving it here, we're just ballooning this file (instead of `ServerController`.

Reply:
I believe this is the best place for all of these validation utils. Every one of these validation methods is directly tied to an OtpServer instance.

Reply:
- `verifyS3WritePermissions` could be generalized and moved to S3Utils (doesn't need to be OtpServer specific).
- `isEmpty` could be moved to `Utils` or some other
- `validateAmiId`, `validateKeyName`, `validateSecurityGroupId` could be generalized and moved to EC2Utils (where there are already some other validation methods)
- `validateIamInstanceProfileArn` could be moved to IAMUtils

Reply:
Moved all of this in 5d3a85bad9f911fbc3f83fccd80bb476f5a5f511